gst/nnstreamer/include/tensor_if.h

```diff
@@ -0,0 +1,58 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer/NNStreamer Tensor-IF
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	nnstreamer_plugin_api_if.h
+ * @date	28 Oct 2020
+ * @brief	NNStreamer APIs for tensor_if custom condition
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * How To for NNdevelopers:
+ *
+ * Define the function in the app.
+ * 1. Define struct, "tensor_if_custom", with the functions defined.
+ * 2. Register the struct with "nnstreamer_if_custom_register" API.
+ * 3. Construct the nnstreamer pipeline and execute it in the app.
+ */
+
+#ifndef __NNS_TENSOR_IF_H__
+#define __NNS_TENSOR_IF_H__
+
+#include <glib.h>
+#include <gst/gst.h>
+#include "tensor_typedef.h"
+
+G_BEGIN_DECLS
+/**
+ * @brief Calls the user defined conditions function.
+ * @param[in] info Input tensors info.
+ * @param[in] input The array of input tensor data.
+ * @param[out] result result of the user defined condition.
+ * @note The maximum array size of input data is NNS_TENSOR_SIZE_LIMIT.
+ * @return TRUE if there is no error.
+ */
+typedef gboolean (* tensor_if_custom) (const GstTensorsInfo *info,
+    const GstTensorMemory *input, gboolean *result);
+
+/**
+ * @brief Register the custom callback function.
+ * @param[in] tensorif_h tensor_if element from the pipeline
+ * @param[in] func The custom condition function body
+ * @param[in/out] data The internal data for the function
+ */
+extern void
+nnstreamer_if_custom_register (GstElement *tensorif_h, tensor_if_custom func, void *data);
```

Comment:
IMO, there are so many name of `tensor if`, like `nnstreamer_if`, `tensorif`, `tensor_if`. How about making a unified name?

Reply:
I rearranged the name of the `tensor_if`. Thank you!